docs/cartesian-product.adoc

```diff
@@ -119,6 +121,110 @@ To demonstrate with a table:
 | 6th test  | 4                 | "B"
 |===
 
+=== `@CartesianEnumSource`
+
+`@CartesianEnumSource` provides a convenient way to use `Enum` constants.
+
+[source,java]
+----
+@CartesianProductTest
+@CartesianEnumSource(ChronoUnit.class)
+void testWithEnumSource(TemporalUnit unit) {
+    assertNotNull(unit);
+}
+----
+
+Like `@CartesianValueSource`, `@CartesianEnumSource` is an annotation that might look familiar - it mimics JUnit's `@EnumSource`, except `@CartesianEnumSource` is repeatable.
+It also does NOT work with `@ParameterizedTest`.
+https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources-EnumSource::[You can read about `@EnumSource` here.]
+
+The annotation’s `value` attribute is optional.
+When omitted, the declared type of the first method parameter is used.
+The test will fail if it does not reference an `Enum` type.
+Thus, the `value` attribute is required in the above example because the method parameter is declared as `TemporalUnit`, i.e. the interface implemented by `ChronoUnit`, which isn’t an enum type.
+Changing the method parameter type to `ChronoUnit` allows you to omit the explicit enum type from the annotation as follows.
+
+[source,java]
+----
+@CartesianProductTest
+@CartesianEnumSource
+void testWithCartesianEnumSourceWithAutoDetection(ChronoUnit unit) {
```

Comment:
Could you add a unit test for this?

Reply:
I thought I did it together with the last doc changes, but I clearly forgot one point 🤦 

https://github.com/junit-pioneer/junit-pioneer/blob/273035edb6c9fe7cb63a69932b5eaf58545639cc/src/test/java/org/junitpioneer/jupiter/CartesianProductTestExtensionTests.java#L934-L938

Reply:
Most importantly, the docs were inspired by the latest version of JUnit (5.7), while the implementation comes from the version included by junit-pioneer (5.5)... still some work is needed :thinking:

Reply:
As mentioned in the other comment, I removed any reference to the auto-detection feature